### ISSUE TITLE
Trouble keeping WebAnno running

### ISSUE THREAD

**Issue by gpythomas** [webanno/webanno#1043]
Hello,

I have issues keeping the WebAnno service running on my server (Ubuntu 16.04). I use the .jar version with embedded Tomcat, and installed it following the documentation guidelines. The service starts fine, but if left alone stops working after a while. For instance, here is the output of `sudo service webanno status` after restarting the service today at 15:06:50 EDT:

```
● webanno.service - LSB: WebAnno - Application
   Loaded: loaded (/etc/init.d/webanno; bad; vendor preset: enabled)
   Active: active (running) since Thu 2018-09-20 15:06:50 EDT; 5s ago
     Docs: man:systemd-sysv-generator(8)
  Process: 5243 ExecStop=/etc/init.d/webanno stop (code=exited, status=0/SUCCESS)
  Process: 5269 ExecStart=/etc/init.d/webanno start (code=exited, status=0/SUCCESS)
   CGroup: /system.slice/webanno.service
           └─5294 /usr/bin/java -Dsun.misc.URLClassPath.disableJarChecking=true -Djava.awt.headless=true -Dwebanno.home=/srv/webanno -jar /srv/webanno/webanno.jar

```
And here it is 12 minutes afterwards:

```
● webanno.service - LSB: WebAnno - Application
   Loaded: loaded (/etc/init.d/webanno; bad; vendor preset: enabled)
   Active: active (exited) since Thu 2018-09-20 15:06:50 EDT; 12min ago
     Docs: man:systemd-sysv-generator(8)
  Process: 5243 ExecStop=/etc/init.d/webanno stop (code=exited, status=0/SUCCESS)
  Process: 5269 ExecStart=/etc/init.d/webanno start (code=exited, status=0/SUCCESS)

Sep 20 15:06:50 linguo systemd[1]: Starting LSB: WebAnno - Application...
Sep 20 15:06:50 linguo webanno[5269]: Started [5294]
Sep 20 15:06:50 linguo systemd[1]: Started LSB: WebAnno - Application.
```
I'm not sure how I should troubleshoot this.

**Comment by reckart** [comment]
Look to me like it didn't start at all:

```
Active: active (running) since Thu 2018-09-20 15:06:50 EDT; 5s ago
Active: active (exited) since Thu 2018-09-20 15:06:50 EDT; 12min ago
```

* Check `/var/log/webanno.log`
* Try running WebAnno manually e.g. using the command that seems to be used by the service wrapper `/usr/bin/java -Dsun.misc.URLClassPath.disableJarChecking=true -Djava.awt.headless=true -Dwebanno.home=/srv/webanno -jar /srv/webanno/webanno.jar`

**Comment by gpythomas** [comment]
At 15:06:50 it was running, or at least I could login to WebAnno after I started it, for a couple of minutes. Then I logged out and when I tried to log back again the application had exited. Here is the content of `/var/log/webanno.log` at startup time:

```
2018-09-20 15:06:52 INFO [SYSTEM] WebAnnoApplicationContextInitializer - Settings: /srv/webanno/settings.properties
2018-09-20 15:06:52 INFO [SYSTEM] WebAnnoApplicationContextInitializer - Authentication: database
2018-09-20 15:06:52 INFO [SYSTEM] WebAnno - Starting WebAnno on linguo with PID 5294 (/srv/webanno/webanno.jar started by www-data in /srv/webanno)
2018-09-20 15:06:52 INFO [SYSTEM] WebAnno - The following profiles are active: auto-mode-builtin
2018-09-20 15:07:08 INFO [SYSTEM] DocumentServiceImpl - Document repository path: /srv/webanno/repository
2018-09-20 15:07:08 INFO [SYSTEM] DatabaseInfoService - Database dialect: org.hibernate.dialect.MySQL5InnoDBDialect
2018-09-20 15:07:08 INFO [SYSTEM] DatabaseInfoService - Database driver: com.mysql.jdbc.Driver
2018-09-20 15:07:08 INFO [SYSTEM] DatabaseInfoService - Database URL: jdbc:mysql://localhost:3306/webanno?useSSL=false
2018-09-20 15:07:08 INFO [SYSTEM] DatabaseInfoService - Database username: webanno
(etc)
```
It looks like it stopped about two minutes afterwards, the session ends like this in the log file:

```
2018-09-20 15:08:09 INFO [admin] AnnotationPage - END LOAD_DOCUMENT_ACTION
2018-09-20 15:08:10 WARN [admin] ResourceReferenceRegistry - A ResourceReference wont be created for a resource with key [scope: de.tudarmstadt.ukp.clarin.webanno.brat.resource.BratCssUiReference; name: static/fonts/Astloch-Bold.ttf; loc$

```
Then I restarted WebAnno at 15:30:18, and it's running since then:

```
2018-09-20 15:30:18 INFO [SYSTEM] WebAnnoApplicationContextInitializer - Settings: /srv/webanno/settings.properties
2018-09-20 15:30:18 INFO [SYSTEM] WebAnnoApplicationContextInitializer - Authentication: database
2018-09-20 15:30:18 INFO [SYSTEM] WebAnno - Starting WebAnno on linguo with PID 8746 (/srv/webanno/webanno.jar started by www-data in /srv/webanno)
```

**Comment by reckart** [comment]
@gpythomas You might want to check your system logs (not from WebAnno but from the OS) to check if possible the kernel out-of-memory-process-killer has killed the Java process. But that shouldn't really happen unless your system is *really* tight on resources.

**Comment by gpythomas** [comment]
I haven't been able to find any clue as to why webanno stops in /var/log/syslog or /var/log/webanno.log (no mention of killed processes in /var/log/syslog, around the time when I know the service exited). I have adopted a quick fix by creating a cron job that checks whether the service is still running and active every 15 minutes, and I have been able to work like this. But the service does seem to stop occasionally. Could this be somehow related to the fact that I don't have any systemd unit file for webanno in /lib/systemd/system or /etc/systemd/system, only a link to webanno.jar in init.d?
